src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsEntityTypeDefinition.cs

```diff
@@ -112,7 +112,36 @@ private IEnumerable<IEdmStructuralProperty> ComputeDeclaredKey()
                 List<IEdmStructuralProperty> key = new List<IEdmStructuralProperty>();
                 foreach (CsdlPropertyReference keyProperty in this.entity.Key.Properties)
                 {
-                    IEdmStructuralProperty structuralProperty = this.FindProperty(keyProperty.PropertyName) as IEdmStructuralProperty;
+                    IEdmStructuralProperty structuralProperty = null;
+                    if (!string.IsNullOrEmpty(keyProperty?.Alias))
+                    {
+                        string[] keyPropertySegments = keyProperty.PropertyName.Split('/');
+                        if (keyPropertySegments.Length > 0)
+                        {
+                            structuralProperty = this.FindProperty(keyPropertySegments[0]) as IEdmStructuralProperty;
+                            for (int i = 1; i < keyPropertySegments.Length; i++)
+                            {
+                                if (structuralProperty.Type.Definition.TypeKind == EdmTypeKind.Complex)
+                                {
+                                    structuralProperty = structuralProperty.Type.AsComplex().FindProperty(keyPropertySegments[i]) as IEdmStructuralProperty;
+                                }
+                                else if (structuralProperty.Type.Definition.TypeKind == EdmTypeKind.Entity)

```

Comment:
I thought the segments in the key property can only reference properties declared using primitive or complex type...
Would this imply that a segment can reference a navigational property?

Reply:
correct.